app/models/form/lettings/pages/person_age.rb

```diff
@@ -1,15 +1,21 @@
 class Form::Lettings::Pages::PersonAge < ::Form::Page
-  def initialize(id, hsh, subsection, person_index:)
+  def initialize(id, hsh, subsection, person_index:, is_child:)
     super(id, hsh, subsection)
-    @id = "person_#{person_index}_age"
-    @depends_on = [{ "details_known_#{person_index}" => 0 }]
+    @id = is_child ? "person_#{person_index}_age_child" : "person_#{person_index}_age_non_child"
```

Comment:
Would it make sense to pass the ID through when initializing?

Reply:
do you mean pass it up the super? I believe that I have to pass something up and if I assign to @id after that line, then it will be overwritten anyway.
I agree this looks a little odd to be passing up something that we happen to know is nil but is called id, but I think it's consistent with most other questions? In most places we pass id (=> nil) implicitly into the super and then assign @id on the next line.